kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/MessageFetchContext.java

```diff
@@ -244,156 +247,179 @@ public void handleFetch() {
             final long offset = partitionData.fetchOffset;
 
             final String fullTopicName = KopTopic.toString(topicPartition);
-            // the future that is returned by getTopicConsumerManager is always completed normally
-            topicManager.getTopicConsumerManager(fullTopicName).thenAccept(tcm -> {
-                if (tcm == null) {
```

Comment:
As we add more implementation details to `handleFetch`, this method becomes too huge. There're many lambda blocks that reference outer local variables or `MessageFetchContext`'s fields. It's hard to read and maintain in future.

Could you add more methods to reduce lines count of a single method?

For example, add a `handlePartitionData` method:

```java
    private void handlePartitionData(final TopicPartition topicPartition,
                                     final FetchRequest.PartitionData partitionData,
                                     final String fullTopicName,
                                     final long startPrepareMetadataNanos,
                                     final boolean readCommitted) {
        final long offset = partitionData.fetchOffset;

        // the future that is returned by getTopicConsumerManager is always completed normally
        topicManager.getTopicConsumerManager(fullTopicName).thenAccept(tcm -> { /* ... */ });
    }
```

Then the `handleFetch` method will become:

```java
    public void handleFetch() {
        final boolean readCommitted =
                (tc != null && fetchRequest.isolationLevel().equals(IsolationLevel.READ_COMMITTED));

        fetchRequest.fetchData().forEach((topicPartition, partitionData) -> {
            final long startPrepareMetadataNanos = MathUtils.nowInNano();
            final String fullTopicName = KopTopic.toString(topicPartition);

            // Do authorization
            requestHandler.authorize(AclOperation.READ, Resource.of(ResourceType.TOPIC, fullTopicName))
                    .whenComplete((isAuthorized, ex) -> {
                        /* handle isAuthorized and ex ... */
                        handlePartitionData(topicPartition, partitionData, fullTopicName, startPrepareMetadataNanos, readCommitted);
                    });
        });
    }
```

Similarly, we can add 

```java
    private void handleEntries(final List<Entry> entries,
                               final TopicPartition topicPartition,
                               final String fullTopicName,
                               final long offset,
                               final KafkaTopicConsumerManager tcm,
                               final ManagedCursor cursor,
                               final AtomicLong cursorOffset,
                               final boolean readCommitted) {
```

Reply:
BTW, there's the same problem with `KafkaRequestHandler#handleTopicMetadataRequest`. When you have time, you can refactor the method like this.

Reply:
Good idea, I'll fix it.